### PR TITLE
MCPimprovements

### PR DESCRIPTION
**Pull request by johnymontana**
mprove the neo4j-agent-memory MCP with a comprehensive 16-tool Claude Desktop-optimized experience with server instructions, tool profiles, automatic memory capture, observational memory, and a high-level `MemoryIntegration` convenience layer.

---

### New: MemoryIntegration Convenience Layer (`integration.py`)

- High-level wrapper over `MemoryClient` for simplified memory operations
- Three session identity strategies: `per_conversation`, `per_day`, `persistent`
- Automatic preference detection from user messages via regex-based `PreferenceDetector`
- Observational memory via `MemoryObserver` for context compression when token thresholds are exceeded
- Returns JSON-serializable dicts (not domain models) for easy consumption by MCP tools and HTTP APIs
- Designed as the shared interface for both MCP server and create-context-graph projects

### New: 16 MCP Tools in Core/Extended Profiles

The old 6 multiplexed tools are replaced with 16 goal-oriented tools.

**Core profile (6 tools):**
`memory_search`, `memory_get_context`, `memory_store_message`, `memory_add_entity`, `memory_add_preference`, `memory_add_fact`

**Extended profile adds 10 more:**
`memory_get_conversation`, `memory_list_sessions`, `memory_get_entity`, `memory_export_graph`, `memory_create_relationship`, `memory_start_trace`, `memory_record_step`, `memory_complete_trace`, `memory_get_observations`, `graph_query`

Select via `--profile core` or `--profile extended` (default).

### New: Server Instructions, Prompts, Resources

- Server instructions sent during MCP initialization guide the LLM on when to call each tool
- 3 prompts: `memory-conversation`, `memory-reasoning`, `memory-review` (slash commands in Claude Desktop)
- 4 resources: `memory://context/{session_id}`, `memory://entities`, `memory://preferences`, `memory://graph/stats`
- All tool annotations set (`readOnlyHint`, `destructiveHint`, `idempotentHint`)

### New: Automatic Memory Capture

- `memory_store_message` auto-triggers entity extraction and background preference detection
- `PreferenceDetector` identifies "I love...", "I don't like...", "my favorite is..." patterns
- `MemoryObserver` tracks accumulated context per session, generates reflections when token threshold exceeded
- Configurable via `--observation-threshold` and `--no-auto-preferences` CLI flags

### New: CLI and Desktop Extension

- `neo4j-memory mcp serve` subcommand with all configuration flags
- `.mcpb` desktop extension manifest in `deploy/mcpb/`
- Full CLI options: `--profile`, `--session-strategy`, `--user-id`, `--transport`, `--observation-threshold`

### New: Documentation

- **Tutorial:** `docs/tutorials/mcp-server.adoc` — step-by-step Claude Desktop setup
- **Reference:** `docs/reference/mcp-tools.adoc` — all 16 tools with parameter tables
- **How-to:** `docs/how-to/create-context-graph-mcp.adoc` — MCP + create-context-graph integration
- Antora nav updated with all new pages

### Test Suite Improvements

- 53 new unit tests: tool error paths, instructions, preference detector edge cases, observer tiers, resource errors
- 18 new integration tests: all 16 tools tested against real Neo4j, `MemoryIntegration` round-trips
- 2 new E2E tests: full conversation lifecycle (13 steps) and reasoning trace lifecycle (6 steps)
- MCP module coverage: **68% → 86%** (`_tools.py` 68% → 91%, `_resources.py` 83% → 100%)
- CI updated: coverage collected from integration tests, merged report job, Codecov flags, 55% threshold
- **1969 tests pass** (900 unit + 290 integration + 259 example + 520 docs)

---

## Breaking Changes

MCP tool names changed (MCP clients discover tools dynamically, so no config changes needed):

| Old Tool | New Tool(s) |
|---|---|
| `memory_store` | `memory_store_message`, `memory_add_preference`, `memory_add_fact` |
| `entity_lookup` | `memory_get_entity` |
| `conversation_history` | `memory_get_conversation` |
| `add_reasoning_trace` | `memory_start_trace` + `memory_record_step` + `memory_complete_trace` |

Internal module paths changed (private API):

| Old Path | New Path |
|---|---|
| `neo4j_agent_memory.mcp.tools` | `neo4j_agent_memory.mcp._tools` |
| `neo4j_agent_memory.mcp.handlers` | removed |

**No breaking changes to:** `MemoryClient`, framework integrations, `Neo4jMemoryMCPServer(client)` constructor, or package exports.

---

## Test Plan

- [x] Unit tests pass (900 passed, 3 skipped)
- [x] Integration tests pass against Neo4j testcontainer (290 passed, 160 skipped for optional deps)
- [x] Example tests pass (259 passed, 57 skipped)
- [x] Docs tests pass (520 passed)
- [x] Full suite: **1969 passed, 269 skipped, 0 failures**
- [x] Lint and format checks pass
- [x] MCP tools tested with FastMCP in-memory Client (unit) and real Neo4j (integration)
- [x] E2E flow tested: store → search → entities → context → graph export → reasoning traces
- [x] Manual test: Claude Desktop with `neo4j-memory mcp serve`
- [x] Manual test: Claude Code with MCP tools
- [x] Cross-host: Cursor, VS Code Copilot

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to/integrations/google-cloud.adoc [hunk 1]
```diff
@@ -369,38 +369,47 @@ The Model Context Protocol (MCP) server exposes memory capabilities as tools for
...
 
 # SSE transport (for Cloud Run/HTTP deployment)
-neo4j-memory mcp serve --transport sse --port 8080
+neo4j-agent-memory mcp serve --transport sse --port 8080
 
 # With custom Neo4j connection
-neo4j-memory mcp serve \
+neo4j-agent-memory mcp serve \
   --neo4j-uri bolt://localhost:7687 \
   --neo4j-password secret
```
[the remark this answers is not in the document]
Comment (johnymontana) [reply docs/modules/ROOT/pages/how-to/integrations/google-cloud.adoc]: @copilot apply changes based on [this feedback](https://github.com/neo4j-labs/agent-memory/pull/80#discussion_r3016662318)

File: src/neo4j_agent_memory/integration.py [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+            for pref in detected:
+                sentiment_prefix = "" if pref.sentiment == "positive" else "Dislikes: "
+                await self.client.long_term.add_preference(
+                    category=pref.category,
+                    preference=f"{sentiment_prefix}{pref.preference}",
+                    context=f"Auto-detected from: {pref.source_text[:200]}",
+                    confidence=pref.confidence,
+                    generate_embedding=True,
+                )
+                logger.debug(
```
[the remark this answers is not in the document]
Comment (johnymontana) [reply src/neo4j_agent_memory/integration.py:214]: @copilot apply changes based on [this feedback](https://github.com/neo4j-labs/agent-memory/pull/80#discussion_r3016662289)

File: tests/unit/mcp/test_mcp_cli.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        result = runner.invoke(
+            cli,
+            ["mcp", "serve", "--password", "pw", "--profile", "core"],
+        )
+        assert result.exit_code == 0
+
+    @patch("neo4j_agent_memory.cli.main.asyncio")
+    @patch("neo4j_agent_memory.mcp.server.run_server", new_callable=AsyncMock)
+    def test_serve_passes_session_strategy(self, mock_run_server, mock_asyncio, runner):
+        mock_asyncio.run = lambda _coro: None
```
[the remark this answers is not in the document]
Comment (johnymontana) [reply tests/unit/mcp/test_mcp_cli.py:103]: @copilot apply changes based on [this feedback](https://github.com/neo4j-labs/agent-memory/pull/80#discussion_r3016662245)
